### DISCUSSION TITLE
Python > duckdb.sql()  Error Handling

### DISCUSSION THREAD

**Discussion by Veldbloemen33** [duckdb/duckdb#7994] [Q&A]
Reader,

Could anybody tell how to catch an error  returned from .sql() command?
Maby point me to a location in the documentation?
Probably i overloop something?

example
sql = 'aaa' 
duckdb.sql(sql )

Error handling???

wkr

**Comment by ned2** [comment]
The main trick is working out which exception you need to catch. The DuckDB docs has [a reference of all the available exceptions](https://duckdb.org/docs/api/python/reference) in the API, to help you workout which is appropriate. `ParserException` (which is what your example throws) is a subclass of `ProgrammingError`, which could be a good candidate.
 
```python
import duckdb

try:
    duckdb.sql("blah")
except duckdb.ParserException as error:
    print(f"oh no: {error}") 
```

**Reply by Veldbloemen33** [reply]
Ned,

The solution works fine.

Thx from the east of the Netherlands.

Rob
